### ISSUE TITLE
[Bug]: Keyboard focus is not moving to the info icon and also it is not accessible with keyboard.

### ISSUE THREAD

**Issue by msftedad** [ceolter/ag-grid#10412]
### Link to reproducible scenario

https://codesandbox.io/p/sandbox/custom-inner-header-component-forked-cfpstj?file=%2Fsrc%2Findex.tsx

### Describe the bug

Steps to reproduce 

1. Open URL: [index.tsx - nodebox - CodeSandbox](https://codesandbox.io/p/sandbox/custom-inner-header-component-forked-cfpstj?file=%2fsrc%2findex.tsx) on latest Edge browser. 
2. code pen will appear.
3. Press tab key to reach to the table.
4. Press tab key to reach to the 'Age' column header of the table.
5. Verify whether keyboard focus is moving to the 'info icon' present in the table or not.

Actual behaviour
On pressing tab after 'Age' column header control focus doesn't move to the info icon it moves to the next interactive control i.e. 'country' column header in Windows.

https://github.com/user-attachments/assets/edb15f15-161f-499f-9980-90f2b6e5f338

<img width="968" alt="Image" src="https://github.com/user-attachments/assets/96b08a89-e872-4140-98ee-ff804dbae028" />

Expected behaviour
On pressing tab after 'Age' column header control focus should move to the info icon present beside it in Windows.

More information
Title: Keyboard focus is not moving to the 'info icon' present in the table in Windows.
User Impact:
Keyboard dependent users are not able to perform functionality related to 'info' icon as focus is not moving to the control.

### Version

30.1.0

### Does the issue occur for a specific framework only?

All languages

### Is the issue only observable on a specific browser?

All browsers

**Comment by ava-utting** [comment]
Hi @msftedad ,

Thanks for reaching out.

When using custom header components, developers are fully responsible for handling all user interactions, which includes keyboard navigation. By default, when a custom header component is focused, the focus is applied to the entire header cell rather than any internal elements. 

This is the behavior you're seeing in your example, where the focus moves from the AGE column directly to the COUNTRY column.

This behaviour is expected and documented here: https://www.ag-grid.com/javascript-data-grid/column-headers/#keyboard-navigation
As such, this is not considered a bug and should be implemented as needed by developers using custom components.

Kind Regards,
Ava
